packages/fuels/tests/contracts.rs

```diff
@@ -642,6 +642,105 @@ async fn contract_call_futures_implement_send() -> Result<(), Error> {
             .await
             .unwrap();
     });
+    Ok(())
+}
+
+fn assert_is_revert_containing_msg(msg: &str, error: Error) {
+    assert!(matches!(error, Error::RevertTransactionError(..)));
+    if let Error::RevertTransactionError(error_message, _) = error {
+        assert!(error_message.contains(msg));
+    }
+}
+
+#[tokio::test]
+async fn test_require() -> Result<(), Error> {
+    setup_contract_test!(
+        contract_instance,
+        wallet,
+        "packages/fuels/tests/contracts/require"
+    );
+
+    let contract_methods = contract_instance.methods();
+    {
+        let error = contract_methods
+            .require_primitive()
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+        dbg!(&error); //TODO: remove after the PR review
+                      //
+        assert_is_revert_containing_msg("42", error);
+    }
+    {
+        let error = contract_methods
+            .require_string()
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+        dbg!(&error); //TODO: remove after the PR review
 
+        assert_is_revert_containing_msg("fuel", error);
+    }
+    {
+        let error = contract_methods
+            .require_custom_generic()
+            .call()
+            .await
+            .expect_err("Should return a revert error");
+        dbg!(&error); //TODO: remove after the PR review
+
+        assert_is_revert_containing_msg("StructDeeplyNestedGeneric", error);
+    }
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_multi_call_require() -> Result<(), Error> {
+    setup_contract_test!(
+        contract_instance,
+        wallet,
+        "packages/fuels/tests/contracts/require"
+    );
+
+    let contract_methods = contract_instance.methods();
+
+    // The output of the error depends on the order of the contract
```

Comment:
Should this also be mentioned in the book?

Reply:
I had a look. I can not insert it easily anywhere in the docs. Maybe I make a new subsection for using `require` in sway ?

Reply:
On second thought, it may be unnecessary. It is intuitive that the first failed require will be the one to pop up since it stops execution anyways.